docs: usage guide for range fields and skill for writing usage guides (#224)
siemens/ix-docs: .github/skills/component-usage-guidelines/SKILL.md

---
name: component-usage-guidelines

> description: 'Create or update Industrial Experience component usage guides. Use when writing guide.md or usage tabs for docs components, including when to use a component, key options, behavior in context, states, dos and don''ts, related links, a final docs build check, and suggestion-oriented wording that works for Siemens and non-Siemens audiences.'

Reviewer: ```suggestion
description: 'Create or update component usage guides in the Industrial Experience design system. Use when writing guide.md or usage tabs for docs components, including when to use a component, key options, behavior in context, states, dos and don'ts, related links, a final docs build check, and suggestion-oriented wording that works for Siemens and non-Siemens audiences.'
```

argument-hint: Which component needs usage guidance?
---

> # Component Usage Guidelines

Reviewer: ```suggestion
# Component usage guidelines
```

Not sure if it's a difference, but we should stay consistent with our own writing guidelines

## Outcome

Produce usage guidance that:

- explains what the component is for in plain language
- uses a short opening definition in `guide.md` that complements, rather than repeats, the `index.mdx` description
- helps readers decide when to use it and when not to
- includes an anatomy illustration and a numbered annotation list for the most important parts
- documents the most relevant options and behaviors in context
- treats `Behavior in context` primarily as guidance for interaction, overflow, text truncation, placement, and responsiveness
- links to related components and writing or accessibility guidance where relevant
- stays separate from API-heavy or framework-specific implementation details
- gives explicit suggestions and recommendations instead of only describing the component
- remains readable for non-Siemens audiences even when Siemens-specific examples are included
- follows the repo styleguide for section order, formatting, grammar, voice and tone
- uses uploaded Figma specifications when available to ground the guidance in actual design intent

## Procedure

> 1. Inspect the target component folder under `docs/components/<component>/`.

Reviewer: Would I give him the component name? In this case, somewhere should be noted something like this:

------

At the start of each guide, ask the user for:

1. **Component name** (e.g., `date-picker`)
2. **Link to the Figma component** in the iX Figma library — this is used to:
   - Place accurate instances of the component in illustrations (not generic placeholders).
   - Inspect the layer structure to identify anatomy parts.

If either input is missing, prompt the user before proceeding.

2. If the target component folder does not exist, create it and add the base documentation files following nearby component patterns, including `index.mdx`, `guide.md`, and `code.mdx`.
3. Read the existing `index.mdx`, `guide.md`, `code.mdx`, `.gemini/styleguide.md`, and any related files to understand what already exists and which house rules apply.
4. Make sure the short opening definition in `guide.md` is not a restatement of the `index.mdx` description. Use the opening to add usage context, decision support, or primary scenarios instead of repeating the summary line.

> 5. If Figma specifications, screenshots or related design material are provided in the chat interaction, analyze them and use them as evidence for anatomy, variants, states and behavior guidance.

Reviewer: Since for me personally, switching between Figma illus and code is actually the biggest pain, I thought about letting him do that as well 😅 (not sure how good that works though, but even a first draft would help) That's why there's a prerequisite step to set up Figma MCP:

> 1. **Check `mcp.json`** in the project root for a Figma server entry:
>    ```json
>    {
>      "servers": {
>        "figma": {
>          "url": "https://mcp.figma.com/mcp",
>          "type": "http"
>        }
>      }
>    }
>    ```
> 2. **Verify tool access**: Attempt a lightweight Figma MCP call (e.g., list files or read a node) to confirm the connection is live.
> 3. **If Figma MCP is unavailable**: Stop and inform the user:
>    > "The Figma MCP server is not reachable. Please check that:
>    > - `mcp.json` contains the Figma server entry above.
>    > - You have enabled MCP access in your Figma account settings (Account → Security → MCP Access).
>    > Once enabled, restart the MCP session and try again."
> 
> Do not proceed with any Figma illustration steps until the connection is confirmed.

Later in the workflow/procedure, I have this:

> ### Create Figma illustrations
> 
> The dedicated Figma illustrations file is: `https://www.figma.com/file/wEptRgAezDU1z80Cn3eZ0o`
> 
> Every page in this file corresponds to a page in the documentation. Locate or create the page matching `{componentName}`. Each illustration is a separate frame on that page.
> 
> **All frames must use:**
> - Width: `1000px`
> - Background: `color-1` token from the iX library
> - Padding: `56px` on all sides
> - Component instances sourced from the Figma component link provided by the user
> 
> #### Illustration 1 — Anatomy
> 
> - Frame title: `{componentName}-anatomy`
> - Place an instance of the component in the frame.
> - Add annotation arrows pointing to each anatomy part identified in Step 1 (layer inspection). Label each arrow to match the list in the Anatomy section of the guide.
> 
> #### Illustration 2 — States
> 
> - Frame title: `{componentName}-states`
> - Show one instance of the component per state (default, hover, active, disabled, loading, focused — include only states relevant to this component).
> - Place a text label above each state instance.
> 
> #### Additional illustrations (optional)
> 
> Only create additional illustrations for "Dos and Don'ts" items if the point cannot be understood from text alone. If created, follow the same frame spec.
> 
> #### After creating each frame
> 
> Retrieve the shareable link to the frame from the Figma MCP and embed it in the draft markdown using:
> 
> ```markdown
> ![{Descriptive alt text}](link-to-figma-frame)
> ```
> 
> Place the anatomy illustration link in the Anatomy section, and the states illustration link in the States section.

6. Add an anatomy illustration near the top of the guide. If no component-specific anatomy image is available yet, use this dummy Figma illustration: `![Anatomy component](https://www.figma.com/design/wEptRgAezDU1z80Cn3eZ0o/iX-Documentation-illustrations?node-id=7794-3088&t=VdWD2lSX0POwEDXH-4)`.

> 7. Always follow the anatomy illustration with a numbered list of annotations that names the key visible parts, for example `1. Start field`, `2. Range separator`, `3. End field`.

Reviewer: If creation works, I would also let him fetch all images:

> ### Fetch illustration assets
> 
> Run the following command to download PNG versions of the illustrations:
> 
> ```bash
> pnpm update-figma 'docs/components/{componentName}/guide.md'
> ```
> 
> Report success to the user once this completes.

- Numbered annotations directly below the anatomy image that name the visible parts readers need to understand
- `## Variants` for meaningful visual or semantic variants
- `## Options` for authoring choices, labels, placements, targets, or configurable behavior
- `## Behavior in context` focused on interaction, overflow, placement, hierarchy, responsiveness or adjacent content rules
- `## States` for default, hover, focus, disabled, open, loading, error, or similar states
- `## Dos and Don’ts` with concise imperative bullets always starting with "Do" or "Don’t" and balanced between dos and don’ts
- `## Related` for adjacent components or broader guidelines

Keep the ordering consistent whenever the sections exist:

1. Frontmatter
2. Short description and anatomy, without a heading, with image and list
3. Variants
4. Options
5. Behavior in context
6. States
7. Dos and Don’ts
8. Related

If you create or revise `code.mdx` together with `guide.md`, follow this structure:

1. Basic
2. Further examples
3. API for the component
	- Properties
	- Events, if relevant
	- Slots, if relevant

## Decision Points

> 

Reviewer: In my version - and the relevance might depend on the model - the context was defined, see below. Also notice the last 2 bullet points to obtain knowledge from other design systems including element.

### Industrial B2B UX context

iX is a design system for industrial software — think SCADA systems, operator dashboards, condition monitoring, asset management and process control UIs. When writing guides, keep this context in mind:

- Users are often operators, engineers or technicians working under time pressure or in safety-critical environments.
- UIs are typically dense with data; components must support efficient scanning, not decorative layouts.
- Dark and light themes both matter; avoid guidance that only works in one theme.
- Accessibility in industrial environments includes considerations beyond standard WCAG, e.g. high-contrast displays, glare, gloved input.
- Consistency and predictability are more important than novelty — users rely on pattern recognition.
- Use the design-systems MCP (`southleft/design-systems-mcp`) to draw on patterns from established external design systems (e.g. Carbon, Material, Atlassian) when relevant.
- Use the `element-mcp` server to consult our sister design system, Element, for alignment and comparison.

- If the component is used inside forms, tables, layouts, or modals, describe placement and interaction constraints in that context.
- Use `Behavior in context` to explain how users interact with the component, how content behaves when space is limited, where the component should appear, and how it adapts across viewport sizes.
- If the component has predefined action patterns, explain what authors can and cannot customize.
- If copy, labels, or messaging are critical, link to the appropriate language guidelines instead of duplicating them.
- If accessibility guidance materially changes usage, link to the relevant accessibility docs.
- If a section would be speculative because the behavior is not documented or visible in existing sources, omit it and keep the guide factual.
- Always include suggestions. Readers should leave with clear advice, not just a description of possible configurations.
- When Siemens-specific practice is relevant, label it clearly as an example such as "Typically at Siemens...", "In our applications...", or "For our software...".
- When possible, pair Siemens-specific examples with a broader recommendation that also makes sense for teams outside Siemens.

> ## Writing Rules

Reviewer: would it make sense to reference the gemini styleguide instead, to avoid redundancy and inconsistencies? - or shorten it down to a few most relevant, and direct him to the styleguide instead, e.g.

- **Voice and Tone**: Helpful, expert and approachable. You are a seasoned member of the iX team guiding your colleagues collaboratively.
- **Style Adherence**: Strictly follow all rules in `styleguide.md`. It is the single source of truth. If anything in this skill conflicts with `styleguide.md`, the style guide wins.
- **Key Phrases**: Use phrases like "Typically at Siemens…", "As a general rule, we…", "We recommend…" and "In our applications, we…" — never direct commands.
- **Plurality**: Always refer to "users" (plural) and components in lowercase plurals (e.g., "buttons", "date pickers").
- **Language**: Active voice, present simple tense, contractions (e.g., "don't", "isn't").

- Use guidance language such as "We recommend...", "We usually...", "As a general rule...", "Typically at Siemens...", or "In our applications..." instead of neutral catalog text.
- Do not assume the reader works at Siemens. Explain Siemens-specific wording or patterns as one example among others.
- Do not use the abbreviation `ix` or `iX` in Industrial Experience design system content. If it appears in usage-guide wording, remove it and align with content writers if broader cleanup is needed.
- Refer to components in lowercase unless the name is an official proper noun. Write "button", "map navigation" and "breadcrumb", not "Button", "Map navigation" or "Breadcrumb". The recognized proper noun exception is "About and legal information".
- Prefer "e.g." over "for example", "such as", or "like".
- Prefer "changelog" over "change log".
- Prefer "among other things" over "inter alia".
- When naming code options in prose, keep the human-readable label aligned with the code spelling from the code section. For example, if code uses `i18nPlainText`, write "Plain text" in the description.
- Remove trailing whitespace from lines.

## What To Avoid

> 

Reviewer: - Never add an `## Anatomy` headline. The anatomy section starts directly with body text.
- Never write placeholder text for empty sections. Omit the section entirely.
- Always use the Figma MCP to create illustrations — never describe them in prose.
- Always source component instances from the user-provided Figma component link, not from generic shapes.
- Always check Figma MCP availability before starting (see §3.1).
- The `Related` section links to component folders, not to specific files like `code.md`.
- Unicode characters per `styleguide.md`: use ' (U+2019), … (U+2026), " (U+201C), " (U+201D).

---
name: component-usage-guidelines

> description: 'Create or update component usage guides in the Industrial Experience design system. Use when writing guide.md or usage tabs for docs components, including when to use a component, key options, behavior in context, states, dos and don''ts, related links, a final docs build check, and suggestion-oriented wording that works for Siemens and non-Siemens audiences.'

Reviewer 2: ```suggestion
description: 'Create or update component usage guides in the Industrial Experience design system. Use when writing guide.md or usage tabs for docs components, including when to use a component, key options, behavior in context, states, dos and don'ts, related links, a final docs build check, and suggestion-oriented wording that works for Siemens and non-Siemens audiences.'
```

```json
{
	"servers": {
		"figma": {
			"url": "https://mcp.figma.com/mcp",
			"type": "http"
		}
	}
}
```

- Verify tool access with a lightweight Figma MCP call before proceeding with illustration work.
- Prefer a lightweight read operation such as reading metadata for the provided Figma component link or another minimal node inspection.
- If Figma MCP is unavailable, stop and tell the user:

	"The Figma MCP server is not reachable. Please check that:

	- `mcp.json` contains the Figma server entry above if your workspace uses project-level MCP configuration
	- MCP access is enabled in your Figma account settings under Account → Security → MCP Access

	Once enabled, restart the MCP session and try again."

Do not proceed with any Figma illustration steps until the connection is confirmed.

## When To Use

- Writing a new `guide.md` or usage tab for a component under `docs/components/`
- Creating a new component documentation folder when the target component does not exist yet
- Refreshing an existing component guide to match current documentation patterns
- Adding missing usage guidance, states, variants, or dos and don’ts

> - Clarifying when to use one component instead of a related alternative

Reviewer 2: not sure this will be clear enoough - 
```suggestion
- Clarifying when to use one component instead of a more appropriate/better alternative
```

Author: ok

- Analyzing Figma specifications provided in the chat interaction and translating them into usage guidance

Do not use this skill for general overview pages unless the task is specifically to write or revise a component usage page.

## Outcome

Produce usage guidance that:

> - explains what the component is for in plain language

Reviewer 2: maybe not clear enough? 
```suggestion
- explains what the component is for in plain language aimed at a grade 8 reading level
```

Author: ok

- uses a short opening definition in `guide.md` that complements, rather than repeats, the `index.mdx` description
- helps readers decide when to use it and when not to
- includes an anatomy illustration and a numbered annotation list for the most important parts
- documents the most relevant options and behaviors in context
- treats `Behavior in context` primarily as guidance for interaction, overflow, text truncation, placement, and responsiveness

> - links to related components and writing or accessibility guidance where relevant

Reviewer 2: ```suggestion
- links to related components and UX writing or accessibility guidance where relevant
```

Author: ok

2. Short description and anatomy, without a heading, with image and list
3. Variants
4. Options
5. Behavior in context
6. States
7. Dos and Don’ts
8. Related

If you create or revise `code.mdx` together with `guide.md`, follow this structure:

1. Basic
2. Further examples
3. API for the component
	- Properties
	- Events, if relevant
	- Slots, if relevant

## Decision Points

- If the component is often confused with another one, explicitly explain when to choose each option.
- If the component is navigational, distinguish navigation from actions.
- If the component is used inside forms, tables, layouts, or modals, describe placement and interaction constraints in that context.
- Use `Behavior in context` to explain how users interact with the component, how content behaves when space is limited, where the component should appear, and how it adapts across viewport sizes.
- If the component has predefined action patterns, explain what authors can and cannot customize.
- If copy, labels, or messaging are critical, link to the appropriate language guidelines instead of duplicating them.
- If accessibility guidance materially changes usage, link to the relevant accessibility docs.

> - If a section would be speculative because the behavior is not documented or visible in existing sources, omit it and keep the guide factual.

Reviewer 2: ```suggestion
- If a section is speculative because the behavior is not documented or visible in existing sources, omit it and keep the guide factual.
```

Author: ok
